### DISCUSSION TITLE
What other Windows capabilities should we consider?

### DISCUSSION THREAD

**Discussion by peterbom** [Azure/aks-periscope#195] [General]
From #168, we have now already:
- Added Windows support for all collectors that aren't Linux-specific, including arbitrary file-system logs
- Enabled OSM/SMI collectors
- Included output from `collect-windows-logs.ps1` (for supported clusters)

There will probably be diminishing returns for the work involved in further Windows capabilities, and so these may not be addressed in the near term but should be noted for future consideration. These include:
- Kubelet collector (which analyses the command line arguments used to invoke the kubelet). This was originally noted as probably being unachievable in Windows, but with host-process containers is likely now achievable (or even trivial).
- The Windows equivalent of the `DnsCollector` which reads `resolv.conf` (in both the container and host). However, I haven't fully explored the `collect-windows-logs` output, and it's possible this contains enough DNS information already.
- A Windows equivalent of the `iptables` output. Again, something like this might already be in the `collect-windows-logs` output.

Any more ideas can be added here.

**Comment by sumicalbin** [comment]
@peterbom One thing we noticed on a large cluster, the data can be overwhelming on what is collected. 

If we wanted to just collect windows logs or networkoutbond logs, etc. Is it possible to customize the yaml and only collect data from the provided collector?
![image](https://user-images.githubusercontent.com/102391206/224982235-62520174-1bb9-4a88-ba26-70338b201767.png)

**Reply by peterbom** [reply]
@sumicalbin - that's not currently possible. The philosophy of Periscope is to collect as much information as possible, as the tool itself can't predict which information will be relevant for troubleshooting.

Also, [the script we run](https://github.com/Azure/aks-engine/blob/master/scripts/collect-windows-logs.ps1) is not maintained by us, and from chatting with @AbelHu I don't believe there are any plans for selectively specifying which logs to collect. One alternative Abel suggested was the [`loggenerator.ps1`](https://github.com/Azure/AgentBaker/blob/master/staging/cse/windows/provisioningscripts/loggenerator.ps1) script. This doesn't call `collectlogs.ps1` and I haven't yet had a chance to understand all the differences, or when you might want to use each. But if it's helpful, one possible enhancement to Periscope might be to allow users to specify _which_ script gets run.

**Reply by sumicalbin** [reply]
I see, thanks for the explanation. It would definitely be a great enhancement to see down the line.
